pyccel/codegen/printing/ccode.py

```diff
@@ -822,13 +822,47 @@ def _print_PythonRound(self, expr):
     def _print_PythonMinMax(self, expr):
         arg = expr.args[0]
         primitive_type = arg.dtype.primitive_type
+        variadic_args = False
+        can_compare = True
 
-        if primitive_type is PrimitiveFloatingPointType() and len(arg) == 2:
+        if isinstance(primitive_type, (PrimitiveFloatingPointType, PrimitiveIntegerType)):
+            variadic_args = True
+
+        if primitive_type is PrimitiveComplexType():
+            can_compare = False
+
+        if isinstance(arg, Variable) and isinstance(arg.class_type , HomogeneousTupleType):
+            arg = PythonTuple(*arg)
```

Comment:
This is not valid. A `HomogeneousTupleType` can have unknown size. In this case `*arg` will return an infinite iterator.

Examples where `HomogeneousTupleType` has unknown size:
```
def f(tup : 'tuple[int,...]'):
    ...

def f(n : int):
    tup = (1, 2, 3)
    longer_tup = tup * n
```

Reply:
I do not know how to check if the size is known at compile time. Could you please give me a hint?

Reply:
The size is known if `shape[0]` is a literal

Reply:
Does this mean that this implementation only works for tuples/lists of known size?

Reply:
What about explicitly sized tuples? Does it support:
```python
def f(tup : 'tuple[int, int, int]'):
    return max(*tup)
```

Reply:
Hmm, that's a shame as it means that there is no way to call min/max on a tuple that is an argument. Couldn't min/max for tuples be mapped to `NumpyAmin`/`NumpyAmax` in the implementation?

Reply:
Sure they can, I will do it through `cspan` extensions; but it is out of the scope of this PR I think. I will dedicate a separate PR to handle homogeneous tuples.